src/server/pps/server/pipeline_controller.go

```diff
@@ -397,50 +384,49 @@ func (pc *pipelineController) loadLatestPipelineInfo(message *pps.PipelineInfo)
 // rcIsFresh returns a boolean indicating whether pc.rc has the right labels
 // corresponding to pc.pipelineInfo. If this returns false, it likely means the
 // current RC is using e.g. an old spec commit or something.
-func (step *pcStep) rcIsFresh() bool {
-	if step.rc == nil {
-		log.Errorf("PPS master: RC for %q is nil", step.pipelineInfo.Pipeline.Name)
+func rcIsFresh(pi *pps.PipelineInfo, rc *v1.ReplicationController) bool {
+	if rc == nil {
+		log.Errorf("PPS master: RC for %q is nil", pi.Pipeline.Name)
 		return false
 	}
-	if step.pipelineInfo == nil {
+	if pi == nil {
 		return true
 	}
-
-	expectedName := ppsutil.PipelineRcName(step.pipelineInfo.Pipeline.Name, step.pipelineInfo.Version)
+	expectedName := ppsutil.PipelineRcName(pi.Pipeline.Name, pi.Version)
 	// establish current RC properties
-	rcName := step.rc.ObjectMeta.Name
-	rcPachVersion := step.rc.ObjectMeta.Annotations[pachVersionAnnotation]
-	rcAuthTokenHash := step.rc.ObjectMeta.Annotations[hashedAuthTokenAnnotation]
-	rcPipelineVersion := step.rc.ObjectMeta.Annotations[pipelineVersionAnnotation]
-	rcSpecCommit := step.rc.ObjectMeta.Annotations[pipelineSpecCommitAnnotation]
+	rcName := rc.ObjectMeta.Name
+	rcPachVersion := rc.ObjectMeta.Annotations[pachVersionAnnotation]
+	rcAuthTokenHash := rc.ObjectMeta.Annotations[hashedAuthTokenAnnotation]
+	rcPipelineVersion := rc.ObjectMeta.Annotations[pipelineVersionAnnotation]
+	rcSpecCommit := rc.ObjectMeta.Annotations[pipelineSpecCommitAnnotation]
 	switch {
-	case rcAuthTokenHash != hashAuthToken(step.pipelineInfo.AuthToken):
+	case rcAuthTokenHash != hashAuthToken(pi.AuthToken):
 		log.Errorf("PPS master: auth token in %q is stale %s != %s",
-			step.pipelineInfo.Pipeline.Name, rcAuthTokenHash, hashAuthToken(step.pipelineInfo.AuthToken))
+			pi.Pipeline.Name, rcAuthTokenHash, hashAuthToken(pi.AuthToken))
 		return false
-	case rcPipelineVersion != strconv.FormatUint(step.pipelineInfo.Version, 10):
+	case rcPipelineVersion != strconv.FormatUint(pi.Version, 10):
 		log.Errorf("PPS master: pipeline version in %q looks stale %s != %d",
-			step.pipelineInfo.Pipeline.Name, rcPipelineVersion, step.pipelineInfo.Version)
+			pi.Pipeline.Name, rcPipelineVersion, pi.Version)
 		return false
-	case rcSpecCommit != step.pipelineInfo.SpecCommit.ID:
+	case rcSpecCommit != pi.SpecCommit.ID:
```

Comment:
just for fun, can we move this check first. we had one customer concerned about logging auth tokens, and even though it's a hash, we might as well do it less

Reply:
I'll just move auth check last?